rfcs/066-graph_pipeline/README.md

```diff
@@ -0,0 +1,244 @@
+
+It is worth noting that an analysis of MeSH concepts from the catalogue has revealed data quality issues affecting some concepts, such as mismatches between labels and MeSH identifiers. For example, there are concepts tagged as MeSH but which have a LoC-type identifier, and MeSH terms such as Sanitation with an identifier which does not link back to Sanitation. Rather than building additional logic into the pipeline to clean this data, we will instead add data quality checks (to make sure we only load source nodes for concepts where the label matches the identifier) and flag these issues directly with the cataloguers. While this will result in a delay for some MeSH concepts to be added to the graph, it will avoid some complexity within the pipeline code.
+
+MeSH concepts will initially only be indirectly linked to LoC concepts via `SAME_AS` edges between Wikidata and these sources. However, there is a possibility to extract direct links from the [UMLS metathesaurus](https://www.nlm.nih.gov/research/umls/index.html), which covers both MeSH and LCSH, in a future iteration of the graph pipeline.
+
+## Other considerations
+
+These are other pipeline considerations which apply to the graph in general, independently from the source data.
+
+### Edge directionality
+
+When creating an edge between two nodes in a graph database such as AWS Neptune, the edge is always directed, even if the direction is omitted in the Cypher query. For example, the following query will create a directed edge from node a to node b:
+```
+MATCH
+    (a:NodeA {property_a: property_a}),
+    (b:NodeB {property_b: property_b})
+MERGE (a)-[r:REL]-(b)
+```
+Furthermore, it is undesirable in terms of query performance to run undirected queries, as explained [here](https://docs.aws.amazon.com/neptune/latest/userguide/best-practices-opencypher-directed-edges.html).
+
+However, the current graph data model does include some edges which do not have any inherent directionality and should be queryable from both sides (for example, `SAME_AS` edges between two source concepts from different ontologies). To avoid introducing an artificial hierarchy between ontologies, and given that this will not add too much storage overhead, these relationships will be represented with two directed edges.
+
+### List properties
+
+It is not possible to create array type properties in AWS Neptune (see [here](https://docs.aws.amazon.com/neptune/latest/userguide/bulk-load-tutorial-format-opencypher.html) for an overview of possible data types). Any list properties (such as `alternative_labels`) will be concatenated to a string, with a `||` separator between individual items. While this can be seen as a limitation, we may benefit from better query performance resulting from indexing of these strings in the graph database.
```

Comment:
Would it be valuable to have labels as nodes in their own right, rather than as part of a property of a node?

It could offer an opportunity to deal with homonyms, e.g.
```mermaid
graph
    A[n79045467]
    B[no95051310]
    C(John Nash)


    B -->|has label| C

    A -->|has label| D(Nash, John, 1752-1835)
    A -->|has label| C
    B -->|has label| E(Nash, John P., 1915-)
```

Reply:
This is an interesting possibility to consider for the future. I think the plan is to include alternative labels in a label-matching or other more sophisticated algorithms either way, but I'm definitely curious in a comparison e.g. from a query performance perspective, or whether this might improve linkage between sources (although we need to be careful about mixing names and concepts).

Reply:
Thinking about it, homonyms are a pretty niche requirement and would probably be better served by a deliberate consideration of reused terms.

Most labels and alternative labels from the sources will be unique to the concept in question, even for relatively common names like John Nash (because most personal or place names have a disambiguator such as their occupation, country, years alive or active etc.).

Reply:
> Most labels and alternative labels from the sources will be unique to the concept in question

That's interesting... something I've been curious about